pkgs/applications/office/paperless/default.nix

```diff
@@ -0,0 +1,170 @@
+{ stdenv
+, lib
+, writeScript
+, fetchFromGitHub
+, makeWrapper
+
+, python36
+, file
+, imagemagickBig # includes ghostscript
+, optipng
+, poppler
+, tesseract
+, unpaper
+}:
+
+## Usage
+
+# ${pkgs.paperless}/bin/paperless wraps manage.py
+
+# ${pkgs.paperless}/share/paperless/setup-env.sh can be sourced from a
+# shell script to setup a Paperless environment
+
+# paperless.withConfig is a convenience function to setup a
+# configured Paperless instance. (See ./withConfig.nix)
+
+# For WSGI with gunicorn, use a shell script like this:
+#   source ${paperless}/share/paperless/setup-env.sh
+#   PYTHONPATH=$paperlessSrc gunicorn paperless.wsgi
+
+with lib;
+let
+  paperless = stdenv.mkDerivation rec {
+    name = "paperless-${version}";
+    version = "2.6.0";
+
+    src = fetchFromGitHub {
+      owner = "danielquinn";
+      repo = "paperless";
+      rev = version;
+      sha256 = "160idy0vwzkb134axkiyj3yf85cgwggz0846w99fb7nckhx7v51m";
+    };
+
+    inherit python;
+
+    runtimePackages = with python.pkgs; [
+      dateparser
+      dateutil
+      django
+      django-crispy-forms
+      django-cors-headers
+      django-filter
+      django_extensions
+      djangorestframework
+      factory_boy
+      filemagic
+      fuzzywuzzy
+      gunicorn
+      langdetect
+      pdftotext
+      pillow
+      pyocrWithUserTesseract
+      python-dotenv
+      python-gnupg
+      pytz
+      termcolor
+    ] ++ (optional stdenv.isLinux inotify_simple);
+
+    checkPackages = with python.pkgs; [
+      coveralls
+      flake8
+      pycodestyle
+      pytest
+      pytest-django
+      pytest-env
+      # pytest_xdist # Disabled, broken in nixpkgs
+      pytestcov
+    ];
+
+    pythonEnv      = python.withPackages (const runtimePackages);
+    pythonCheckEnv = python.withPackages (const (runtimePackages ++ checkPackages));
+
+    nativeBuildInputs = [ makeWrapper ];
+
+    phases = [ "unpackPhase" "patchPhase" "buildPhase" "checkPhase" ];
+
+    doCheck = true;
+
+    unpackPhase = ''
+      srcDir=$out/share/paperless
+      mkdir -p $srcDir
+      cp -r --no-preserve=mode $src/src/* $srcDir
+      cp $src/LICENSE $srcDir
+    '';
+
+    patchPhase = ''
+      # Remove pytest command arguments for pytest_xdist, which is broken in nixpkgs
+      sed -i 's/ -n auto//' $srcDir/setup.cfg
+
+      # Skip failing test (see Paperless pull request 450)
```

Comment:
Sorry, I did not really look at the patch. Seems like it won't have a meaningful effect on our side. Instead you should update the comment to state that the disabled tests requires German language support for Tesseract.

Reply:
The German lang support is irrelevant to the test failure.
The patch that fixes the test failure consists of a large `GIT binary patch` that can't be applied with `fetchpatch`.
Because the patch will be included in the next Paperless release, I chose to just skip the test for now.
I've moved the relevant code to `postPatch`: https://github.com/NixOS/nixpkgs/pull/44086/commits/55aa8bda951841b7c87c31cfedb4e2b4a6fe4561